main.py

```diff
@@ -83,11 +104,20 @@ def run(self):
             if PREF.get('distraction_free_window.toggle_status_bar', False):
                 w.set_status_bar_visible(True)
 
-        try:
-            # toggle MaxPane if found
-            PKGCTRL_PREF = sublime.load_settings('Package Control.sublime-settings')
-            is_maxpane_installed = bool('MaxPane' in set(PKGCTRL_PREF.get('installed_packages', [])))
-            if is_maxpane_installed:
-                self.window.run_command('max_pane')
-        except Exception as e:
-            pass
+        if toggle:
+            try:
+                # toggle MaxPane if found
+                PKGCTRL_PREF = sublime.load_settings('Package Control.sublime-settings')
+                is_maxpane_installed = bool('MaxPane' in set(PKGCTRL_PREF.get('installed_packages', [])))
+                if is_maxpane_installed:
+                    self.window.run_command('max_pane')
+            except Exception as e:
+                pass
+
+            try:
+                with open(session_file, "w") as file:
+                    file.write(sublime.encode_value(session_settings, pretty=True))
+            except FileNotFoundError:
+                self._status_msg('Error: Session file not found! Mode was not saved!')
```

Comment:
```python
if session_settings is not None:
    with os.open(session_file, mode='x', newline='\n') as file:
        file.write(sublime.encode_value(session_settings, pretty=True))
```

would create the file, although it doesn't make sense to do that here. Just a reference for interested reviewers.

Reply:
Why not?